File: app/src/main/java/com/arad/care4pets/PetProfileActivity.java
package com.arad.care4pets;

import android.os.Bundle;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

public class PetProfileActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_pet_profile);

        TextView tvName = findViewById(R.id.tvProfileName);
        TextView tvSpecies = findViewById(R.id.tvProfileSpecies);
        TextView tvAge = findViewById(R.id.tvProfileAge);
        TextView tvNotes = findViewById(R.id.tvProfileNotes);

        int petId = getIntent().getIntExtra("petId", -1);

        Pet pet = null;
        for (Pet p : FakeRepository.pets) {
        if (p.getId() == petId) {
            pet = p;
            break;
        }
    }

        if (pet != null) {
            tvName.setText(pet.getName());
            tvSpecies.setText("Species: " + pet.getSpecies());
            tvAge.setText("Age: " + pet.getAge() + " years");
            String notes = pet.getNotes() != null ? pet.getNotes().trim() : "";
            tvNotes.setText(!notes.isEmpty() ? notes : "No special notes");
        }
    }
}

File: app/src/main/java/com/arad/care4pets/PetProfileActivity.kt
package com.arad.care4pets

import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class PetProfileActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_pet_profile)

        val tvName = findViewById<TextView>(R.id.tvProfileName)
        val tvSpecies = findViewById<TextView>(R.id.tvProfileSpecies)
        val tvAge = findViewById<TextView>(R.id.tvProfileAge)
        val tvNotes = findViewById<TextView>(R.id.tvProfileNotes)

        val petId = intent.getIntExtra("petId", -1)
        val pet = FakeRepository.pets.firstOrNull { it.id == petId }

        pet?.let {
            tvName.text = it.name
            tvSpecies.text = "Species: ${it.species}"
            tvAge.text = "Age: ${it.age} years"
            tvNotes.text = if (it.notes.isNotEmpty()) it.notes else "No special notes"
        }
    }
}
